Final review before production

Metsanpeitto opened the pull request:
This is the last merge that will occur before production.
All the tasks have been properly divided into Kanban cards, and those Kanban cards were translated into branches that were submitted to review by other team members.
After the approval of each pull request, those branches were merged into developing, which now holds a totally functional version of the app.

The front end pull request link is this:
[Frontend Pull Request](https://github.com/Metsanpeitto/microverse_final_project_front/pull/27/)

The app has been already deployed on Heroku 
[API Deployed](https://frozen-dusk-66130.herokuapp.com/api-docs/index.html)

Metsanpeitto committed:
First commit

Metsanpeitto committed:
Set up the API, add cors to initializers, add routes and controller ready

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Generate Model and controller for item resource

Metsanpeitto committed:
Generate the model and controller for the reservations resource

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Update initializer cors to accept * as origins

Metsanpeitto committed:
Remove user manager code

Metsanpeitto committed:
Create the devise and devise_jwt connection to the api

Metsanpeitto committed:
Recover the ost reservations_controller

Metsanpeitto committed:
Add seed and fix migration order

Metsanpeitto committed:
Add user role migration and model validation

Metsanpeitto committed:
Update the seed user

Metsanpeitto committed:
Add user_id to parameter validator in item controller

Metsanpeitto committed:
Add dependent: :destroy to the item and reservation models

Metsanpeitto committed:
Remove the dependent: :destroy from reservation model

Metsanpeitto committed:
Let ready the RSpec testing and the Rswagger API documentation

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Merge pull request #11 from Metsanpeitto/create-scaffold-items

Create scaffold items

Metsanpeitto committed:
Merge pull request #12 from Metsanpeitto/create-scaffold-reservations

Create scaffold reservations

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Remove test folder

Metsanpeitto committed:
Merge branch 'develop' into user-manager

Metsanpeitto committed:
Merge pull request #33 from Metsanpeitto/user-manager

Create User manager authentication

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Merge branch 'develop' into api-documentation

Metsanpeitto committed:
Fix linters

Metsanpeitto committed:
Merge pull request #35 from Metsanpeitto/api-documentation

Api documentation

Metsanpeitto committed:
Upload README file to explain how to modify the master key

Metsanpeitto committed:
Add line require_master_key in production.rb

Metsanpeitto committed:
Add root to routes

Metsanpeitto committed:
Update routes

Metsanpeitto committed:
Add Procfile to root folder

Metsanpeitto committed:
Update routes

Metsanpeitto committed:
Update README to add instructions

Metsanpeitto committed:
Fix typo in README

Metsanpeitto committed:
Fis jwt credentials

Metsanpeitto committed:
Fix jwt

Metsanpeitto committed:
Merge branch 'fix-bugs-develop' into develop

Metsanpeitto committed:
Update the README file

Metsanpeitto committed:
Update README

Metsanpeitto committed:
Update README

rvvergara commented on app/controllers/current_user_controller.rb:
```diff
@@ -0,0 +1,6 @@
+class CurrentUserController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    render json: current_user, status: :ok
```
Hi guys. Im curious as to how exactly we're using this and how necessary this method is.

Our API should be stateless, meaning it shouldn't be the one keeping state of a user's authenticated condition. Therefore, having a "current_user" in the backend might not seem intuitive.

Metsanpeitto replied:
:heavy_check_mark: Removed! 
It was not needed, indeed. :+1:

rvvergara replied:
Great!!

rvvergara commented on app/controllers/reservations_controller.rb:
```diff
@@ -0,0 +1,45 @@
+
+  # GET /reservations/1
+  def show
+    render json: @reservation
```
don't you think we should also require only authenticated users to access this endpoint?

Metsanpeitto replied:
:heavy_check_mark: Updated! 
```cmd
  before_action :authenticate_user!, only: %i[create update show]
```
:+1:

rvvergara replied:
Awesome!

rvvergara commented on config/routes.rb:
```diff
@@ -0,0 +1,11 @@
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  root to: "items#index"
+  get 'current_user/index'
```
Ditto in relation to my comment in the current_user controller. How is this endpoint necessary for the app?

Metsanpeitto replied:
:heavy_check_mark: Removed! 
It was not needed, indeed. :+1:

rvvergara reviewed (changes requested):
Hi team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

Two major things I notice:

- I am not sure if this review request is only meant for your back end but I didn't seem to see any link to your front end app for me to review. Please make sure you include a link to it in your succeeding code review request.

- We need to write tests for our API


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Metsanpeitto committed:
Fix bugs and implement changes requested by the code reviewer

Metsanpeitto commented:
@rvvergara  Hola!
Thanks for your time and for pointing out those things. I fixed them already.

However. In my pull request : 
```cmd 
This is the last merge that will occur before production.
All the tasks have been properly divided into Kanban cards, and those Kanban cards were translated into branches that were submitted to review by other team members.
After the approval of each pull request, those branches were merged into developing, which now holds a totally functional version of the app.

The front end pull request link is this:
Frontend Pull Request    <-----    THIS LINK IS THE FRONT END PULL REQUEST 
```
Please visit the link here: 

The front end pull request link is this:
[Frontend Pull Request](https://github.com/Metsanpeitto/microverse_final_project_front/pull/27/)

We have also implemented RSwag documentation that requires RSpec in order to work successfully.
Visit the API documentation in the deployed version : 

### API Documentation

[API Documentation](https://frozen-dusk-66130.herokuapp.com/api-docs/index.html)



The app has been already deployed on Heroku 
[API Deployed](https://frozen-dusk-66130.herokuapp.com/api-docs/index.html)

Metsanpeitto committed:
Update README

rvvergara reviewed (approved):
Hi team, 

In my opinion, your back end is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Metsanpeitto committed:
Update README
